feature/path-tracking/src/main/java/com/mifos/feature/path_tracking/PathTrackingScreen.kt

```diff
@@ -0,0 +1,317 @@
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.MapUiSettings
+import com.google.maps.android.compose.rememberCameraPositionState
+import com.mifos.core.common.utils.Constants
+import com.mifos.core.designsystem.component.MifosCircularProgress
+import com.mifos.core.designsystem.component.MifosScaffold
+import com.mifos.core.designsystem.component.MifosSweetError
+import com.mifos.core.designsystem.icon.MifosIcons
+import com.mifos.core.designsystem.theme.Black
+import com.mifos.core.designsystem.theme.White
+import com.mifos.core.objects.user.UserLatLng
+import com.mifos.core.objects.user.UserLocation
+
+@Composable
+fun PathTrackingScreen(
+    userId: Int,
+    onBackPressed: () -> Unit,
+    onPathTrackingClick: (List<UserLatLng>) -> Unit
+) {
+
+    val context = LocalContext.current
+    val viewModel: PathTrackingViewModel = hiltViewModel()
+    val state by viewModel.pathTrackingUiState.collectAsStateWithLifecycle()
+    val refreshState by viewModel.isRefreshing.collectAsStateWithLifecycle()
+    val userStatus by viewModel.userStatus.collectAsStateWithLifecycle()
+
+    DisposableEffect(Unit) {
+
+        val notificationReceiver = object : BroadcastReceiver() {
+            override fun onReceive(context: Context, intent: Intent) {
+                val action = intent.action
+                if (Constants.STOP_TRACKING == action) {
+                    viewModel.loadPathTracking(userId)
+                }
+            }
+        }
+        registerReceiver(
+            context,
+            notificationReceiver,
+            IntentFilter(Constants.STOP_TRACKING),
+            ContextCompat.RECEIVER_NOT_EXPORTED
+        )
+
+        onDispose {
+            context.unregisterReceiver(notificationReceiver)
+        }
+    }
+
+    LaunchedEffect(Unit) {
+        viewModel.loadPathTracking(userId)
+    }
+
+    PathTrackingScreen(
+        state = state,
+        onBackPressed = onBackPressed,
+        onRetry = {
+            viewModel.loadPathTracking(userId)
+        },
+        onPathTrackingClick = onPathTrackingClick,
+        onRefresh = {
+            viewModel.refreshCenterList(userId)
+        },
+        refreshState = refreshState,
+        userStatus = userStatus,
+        updateUserStatus = { viewModel.updateUserStatus(it) }
+    )
+}
+
+@Composable
+fun PathTrackingScreen(
+    state: PathTrackingUiState,
+    onBackPressed: () -> Unit,
+    onRetry: () -> Unit,
+    onPathTrackingClick: (List<UserLatLng>) -> Unit,
+    onRefresh: () -> Unit,
+    refreshState: Boolean,
+    userStatus: Boolean,
+    updateUserStatus: (Boolean) -> Unit
+) {
+
+    val snackbarHostState = remember { SnackbarHostState() }
+    val pullRefreshState = rememberPullRefreshState(
+        refreshing = refreshState,
+        onRefresh = onRefresh
+    )
+    val permissionState = rememberMultiplePermissionsState(
```

Comment:
Connect with Aditya (Mister K) and implement generic permission.

Reply:
Done